source/main/physics/BeamFactory.cpp

```diff
@@ -608,12 +608,20 @@ void BeamFactory::UpdateSleepingState(float dt)
 	{
 		current_truck->state = SIMULATED;
 	}
+
+	std::bitset<MAX_TRUCKS> visited;
+	// Recursivly activate all trucks which can be reached from current_truck
 	if (current_truck && current_truck->state == SIMULATED)
 	{
 		current_truck->sleeptime = 0.0f;
-		std::bitset<MAX_TRUCKS> visited;
 		this->RecursiveActivation(m_current_truck, visited);
 	}
+	// Snowball effect (activate all trucks which might soon get hit by a moving truck)
+	for (int t=0; t<m_free_truck; t++)
+	{
+		if (m_trucks[t] && m_trucks[t]->state == SIMULATED && m_trucks[t]->sleeptime == 0.0f)
+			this->RecursiveActivation(t, visited);

```

Comment:
I've re-read the code and this is what I got:
- the `RecursiveActivation(truck)` marks the `truck` immediately as visited.

Hence:
 1. `RecursiveActivation(curr_truck)` marks curr_truck visited.
 2. The loop inside will traverse ALL other trucks: https://github.com/ulteq/rigs-of-rods/blob/f45980ccabb3a921be1f283024a7f52dfd72babc/source/main/physics/BeamFactory.cpp#L570 - hence marking them visited